### PR TITLE
Show "Differential expression" button whenever study has DE results (SCP-4954)

### PR DESCRIPTION
**Pull request by eweitz**
This shows the "Differential expression" button in more cases, helping discoverability now that we have 90% less DE data.

### Background <span id="background"></span>
Previously, we [showed the DE button](https://github.com/broadinstitute/single_cell_portal_core/pull/1502) only when the current clustering and annotation had DE results.  Then we [restricted DE](https://github.com/broadinstitute/single_cell_portal_core/pull/1715) to only cell type-like annotations, because the false positive rate of the Wilcoxon method seems too high even for exploratory analysis.  Naively, this would mean showing the DE button 90% less frequently.  That in turn would likely greatly diminish the overall value of DE in SCP.

Now, to A) only surface higher-quality DE while also B) avoiding a precipitous drop in DE discoverability and engagement, the big blue "Differential expression" button is shown whenever _any_ annotation in the study has DE results, even if the current annotation lacks DE results.  This means the DE button will be shown in _more_ cases than before the 90% restriction.

In that 90% case, the UX flow is:
1. See the DE button
2. Select a different annotation (and sometimes also a different clustering) from a list of those that have DE results
3. Select a group 
4. See the DE table

The main UX cost here is the the "Differential expression" button is more for a general feature of the study, rather than something directly related to the current clustering and annotation like the other controls in the right panel.  The bet here is that any cost in user confusion will be substantially less than the benefit of exploring DE results.

### Impact <span id="impact"></span>
The updated DE UX will likely increase indicators of interest (i.e. clicks on the DE button among users who see it), and may well also increase deeper DE engagement, like interacting with the DE table.

### Video <span id="video"></span>
Here's how it looks:

https://user-images.githubusercontent.com/1334561/222259046-5473589b-71cb-41bc-b384-4a5ba17864ac.mov

### Test <span id="test"></span>
A new automated test confirms behavior.  Optionally, if you want to manually verify, follow the steps in the video, i.e.:

1.  **If local**, after pulling, since we're only testing frontend changes, try booting with the frontend service worker cache enabled for faster page loads: `bin/docker-compose-setup.sh -c`.  Then go to a study with DE results.  **If on staging**, go to ["Human milk - differential expression"](https://singlecell-staging.broadinstitute.org/single_cell/study/SCP303/human-milk-differential-expression#study-visualize) (SCP303).
2. Change the default annotation to one that does not have "cell" and "type" in the name (e.g. `biosample_id`).  E.g. click "Settings" then "View options", then change "Default annotation" to e.g. "biosample_id", then click "Update settings", then go back to "Explore".
3. Confirm the blue "Differential expression" button appears, click it
4. Confirm the panel says "No DE results for:", and lists the current clustering and annotation, then gives a dropdown to select a clustering / annotation that has DE.  Select one of them.  If your current clustering lacks DE results, select one from the menu, then select an annotation.
6. Select a group
7. Confirm you see the DE table
8. Confirm the scatter plots show the name of your newly selected clustering and annotation

### Further details
[Slides](https://docs.google.com/presentation/d/1Zd_FO7HqUlxORJqTMEbp5SmZto-PvMKdtTg35nYecS4/edit) give more context.  Recordings for the [demo presentation](https://drive.google.com/file/d/1_6buLOdH4WLqUPMqFTC2QCQvH--DHCRp/view?t=51s) and [feedback](https://drive.google.com/file/d/1_6buLOdH4WLqUPMqFTC2QCQvH--DHCRp/view?t=15m02s) add richer commentary.

This satisfies SCP-4954.

### CODE REVIEW

**Comment by eweitz** [comment]
The [build failure](https://github.com/broadinstitute/single_cell_portal_core/actions/runs/4308224441/jobs/7514244045#step:7:4435) seems like a false positive, as only a Rails test failed but no Rails code changed.

```
### FAILURES and ERRORS ###
*** Yarn test failures ***

*** Rails test failures ***

  AbTestsControllerTest#test_should_destroy_A/B_test [/home/app/webapp/test/controllers/ab_tests_controller_test.rb:49]:
  Expected #<AbTest _id: 63ffcef8421aa901c817e9c7, created_at: 2023-03-01 22:17:28.574 UTC, updated_at: 2023-03-01 22:17:28.574 UTC, feature_flag_id: BSON::ObjectId('63ffcef8421aa901c817e9c4'), group_names: ["control", "intervention"], enabled: false> to be nil.
Exiting with code: 1
Error: Process completed with exit code 1.
```

**Review by bistline — APPROVED** [review]
Looks good, works as advertised.

Question: when testing locally, I experienced this situation:
Screen Shot 2023-03-02 at 10 26 15 AM

I'm assuming that the repeat of `General_Celltype` is because there two clusters in this study?  Is there a way the we can either a) uniquify this list, or b) display that there are results available on the other cluster?  It wasn't immediately obvious to me where this list was being rendered from.

**Review by ehanna4 — CHANGES_REQUESTED** [review]
I am working through functional review but have run into 2 issues
First and the one that I would call blocking is that the Annotation name appears to not come through correctly for a user designated Annotation (see video)

Second not blocking but would be good to be fixed is that when the user annotation is being created the DE button overlaps the spinner a bit, would be good to add some padding or something to separate those a bit (screenshot) (EDIT - added a 2nd screenshot)

https://user-images.githubusercontent.com/54322292/222561559-06bdc486-c445-49c0-b714-74804118cf86.mov

Screen Shot 2023-03-02 at 4 35 05 PM
Screen Shot 2023-03-02 at 4 39 05 PM

File: app/javascript/components/visualization/controls/AnnotationSelector.jsx [hunk 1]
```diff
@@ -4,6 +4,8 @@ import _clone from 'lodash/clone'
 import Select from '~/lib/InstrumentedSelect'
 import { annotationKeyProperties, getMatchedAnnotation, clusterSelectStyle } from '~/lib/cluster-utils'
 
+const noneSelected = 'Select annotation'
```
[the remark this answers is not in the document]
Comment (eweitz) [reply app/javascript/components/visualization/controls/AnnotationSelector.jsx]: No, nice catch!  This was left over from development.

**Review by ehanna4 — APPROVED** [review]
That change fixed it! Thanks, looks good!

**Comment by eweitz** [comment]
Like before, the [build failure](https://github.com/broadinstitute/single_cell_portal_core/actions/runs/4347708226/jobs/7595333690#step:7:4471) seems like a false positive, as only a Rails test failed but no Rails code changed.

```
### FAILURES and ERRORS ###
*** Yarn test failures ***

*** Rails test failures ***

  AbTestsControllerTest#test_should_add_user_to_A/B_test_group:
  NoMethodError: undefined method `group_name' for nil:NilClass

  AbTestsControllerTest#test_should_destroy_A/B_test [/home/app/webapp/test/controllers/ab_tests_controller_test.rb:49]:
  Expected #<AbTest _id: 640651ff421aa901c98ccdef, created_at: 2023-03-06 20:50:07.203 UTC, updated_at: 2023-03-06 20:50:07.203 UTC, feature_flag_id: BSON::ObjectId('640651fe421aa901c98ccde7'), group_names: ["control", "intervention"], enabled: false> to be nil.
Exiting with code: 1
Error: Process completed with exit code 1.
```

Per standup chat, I'll go ahead and merge.  The bug Jon encountered seems incidentally prevented by a separate issue.

**Comment by bistline** [comment]
> Like before, the [build failure](https://github.com/broadinstitute/single_cell_portal_core/actions/runs/4347708226/jobs/7595333690#step:7:4471) seems like a false positive, as only a Rails test failed but no Rails code changed.

We're seeing this on occasion in builds but not consistently.  I was able to reproduce it once locally but the cause was that static assets were not compiled correctly for the `test` environment.  If that was the cause then we'd expect all controller tests to fail, not just this one.  So (sadly) this is a potential lurker but I don't think it's an actual bug.

**Pull request merged**